### PR TITLE
Make IntSet iterator double ended.

### PR DESCRIPTION
**Pull request by garretrieger**
To allow for reverse iteration.

### CODE REVIEW

File: int-set/src/lib.rs [hunk 1]
```diff
@@ -189,6 +190,16 @@ impl<T: Domain<T>> IntSet<T> {
         }
     }
 
+    /// Returns the smallest member of this set if there is one.
+    pub fn min(&self) -> Option<T> {
```
Comment (cmyr) [inline int-set/src/lib.rs:194]: `BTreeSet` calls these fns `first` and `last`, maybe that would make sense? See https://doc.rust-lang.org/1.78.0/std/collections/struct.BTreeSet.html#method.last

File: int-set/src/lib.rs [hunk 2]
```diff
@@ -374,11 +471,33 @@ impl Domain<u16> for u16 {
         true
     }
 
-    fn ordered_values() -> impl Iterator<Item = u32> {
+    fn ordered_values() -> impl DoubleEndedIterator<Item = u32> {
         (u16::MIN as u32)..=(u16::MAX as u32)
     }
 
-    fn ordered_values_range(range: RangeInclusive<u16>) -> impl Iterator<Item = u32> {
+    fn ordered_values_range(range: RangeInclusive<u16>) -> impl DoubleEndedIterator<Item = u32> {
```
Comment (cmyr) [inline int-set/src/lib.rs:478]: at some point I would probably write a macro do to these impls, since they're all so similar.

**Review by cmyr — APPROVED** [review]
looks great, a couple comments inline but nothing blocking :)

File: int-set/src/lib.rs (hunk 1 above)
Comment (garretrieger) [reply int-set/src/lib.rs]: sounds good.

File: int-set/src/lib.rs (hunk 2 above)
Comment (garretrieger) [reply int-set/src/lib.rs:478]: Yeah, sounds good

**Pull request merged**